spec/bolt_server/transport_app_spec.rb

```diff
@@ -787,82 +788,184 @@ def post_over_transport(transport, action, body_content, multiple: false)
+        expect(error).to include("`module_name`: bar does not exist")
         expect(last_response.status).to eq(400)
       end
 
       it 'returns 400 if file does not exist in the module' do
-        allow(fake_module).to receive(:file).with(file).and_return(nil)
-        get(path)
+        get("/project_file_metadatas/project_module/not_a_real_file?project_ref=#{project_ref}")
         error = last_response.body
-        expect(error).to eq("`file`: #{file} does not exist inside the module's 'files' directory")
+        expect(error).to include("`file`: not_a_real_file does not exist inside the module's 'files' directory")
         expect(last_response.status).to eq(400)
       end
 
-      it 'returns the file metadata of the file and all its children' do
-        get(path)
-        file_metadatas = last_response.body
-        expect(file_metadatas).to eq("[]")
-        expect(last_response.status).to eq(200)
+      context "with a valid filepath to one file", ssh: true do
+        let(:test_file) {
+          Pathname.new(
+            File.join(project_dir, project_ref, 'modules', 'project_module', 'files', 'test_file')
+          ).cleanpath.to_s
+        }
+        let(:file_checksum) {
+          Digest::SHA256.hexdigest(
+            File.read(test_file)
+          )
+        }
+        let(:expected_response) {
+          [
+            {
+              "path" => test_file,
+              "relative_path" => ".",
+              "links" => "follow",
+              "owner" => File.stat(test_file).uid,
+              "group" => File.stat(test_file).gid,
+              "checksum" => {
+                "type" => "sha256",
+                "value" => "{sha256}#{file_checksum}"
+              },
+              "type" => "file",
+              "destination" => nil
+            }
+          ]
+        }
+        it 'returns the file metadata of the file and all its children' do
+          get("/project_file_metadatas/project_module/test_file?project_ref=#{project_ref}")
+          file_metadatas = JSON.parse(last_response.body)
+          # I don't know why the mode returned by puppet is not the same as the mode returned
+          # from ruby's File.stat(test_file) function. But these tests probably don't need to
+          # cover the specifics of what puppet returns, plus we don't use this metadata in
+          # orch anyway, so ignore the mode part of the respose.
+          #                                     - Sean P. McDonald 10/15/2020
+          file_metadatas.each do |entry|
+            entry.delete("mode")
+          end
+          expect(file_metadatas).to eq(expected_response)
+          expect(last_response.status).to eq(200)
+        end
       end
 
-      context "when the file path contains '/'" do
-        let(:file) { "foo/bar" }
+      context "when the file path contains '/'", ssh: true do
+        skip "The transport App does not run on windows" if Bolt::Util.windows?
```

Comment:
This message tells me that more tests should be skipped, but we are not doing that. Maybe reword to explain why _this_ test is skipped?

Reply:
Oh, whoops

I'm just going to remove this